uriparser: 0.8.2 -> 0.8.4 + fixed build

c0bw3b opened the pull request:
###### Motivation for this change
Minor version bump + removed the creation of the HTML API documentation because :
1. it is available online on the website of the project
2. it requires `doxygen`, `graphviz` and `qhelpgenerator` which is part of the whole Qt framework...

This should be pushed to `release-17.03` as part of #23253

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

c0bw3b committed:
uriparser: 0.8.2 -> 0.8.4 + fixed build

Minor version bump + removed doc

c0bw3b commented:
Running the bin from the package :

```bash
[demo@nixos:~/testbuild/uriparser]$ ./result/bin/uriparse https://nixos.org/nixos/security.html
uri:          https://nixos.org/nixos/security.html
scheme:       https
hostText:     nixos.org
 .. pathSeg:  nixos
 .. pathSeg:  security.html
absolutePath: false
```

bjornfor commented:
The commit message needs an explanation why removing the docs is a good thing. (It generally isn't :-) )

c0bw3b committed:
uriparser: 0.8.2 -> 0.8.4

Minor version bump.
Fixed the Makefile for the doc : there is no .map files to install in this release since dot outputs images in PNG and SVG

c0bw3b commented:
@bjornfor Fair point.

I've reworked this and figured out what was blocking the build from the doc Makefile.
